# Contrato *Forward rate agreement* (FRA). 


Son compromisos entre dos partes sobre un tipo de interés de un valor teórico (nocional) a un plazo y por un importe 
determinado dado un lapso.
El método para calcular la liquidación a pagar por quien compra o viceversa o es como sigue:

$$L= \frac{(i_{p}- i_{r})\centerdot T\centerdot N}{36000+(i_{p}\centerdot T)}$$

En donde $L$ es el importe de liquidación, quien compra si este es negativo pagará a quien vende y viceversa. $i_{p}$ es el interés garantizado porcentual en FRA. $i_{r}$ es el interés vigente al momento de la operación. $T$ es el número de días de diposición del depósito teórico. $N$ es el valor nominal del contrato.  

In [16]:
def fra (ip,ir,T,N):
    return ((ip-ir)*T*N)/(36000+(ir*T))

Sea un FRA 3/9 meses con depósito de 10,000,000, a tres meses la tasa de referencia es de 5% por 6 meses, con interés pactado del 4%.

In [20]:
fra (4,5,180,10000000)

-48780.48780487805

Es decir, quien vende paga 48780. 

Sea un FRA 3/6 meses con depósito de 500,000,000 a tipo de interés pactado del 8% y el interés refencial a tres meses es de 7%.

In [28]:
fra (8,7,90,500000000)

1228501.2285012286

Es decir, quien compra paga 1228501. La fórmula hasta este momento, sencilla: si el interés pactado es menor al interés al momento de la operación, quien vende paga; si el interés pactado es mayor al interés al momento de la operación, quien compra paga.